### ISSUE TITLE
[Bug]: "The ARIA attributes "aria-label" are not valid for the element <header> with implicit ARIA role "generic", in the UI Shell's header component

### ISSUE THREAD

**Issue by Gururajj77** [carbon-design-system/carbon#19725]
### Package

@carbon/react

### Browser

Chrome

### Package version

v11

### React version

v19

### Description

Context from this [slack thread](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1750333381614129).

The typescript interface doesn't have the enforcement of mandating either `aria-label` or `aria-labelledby`, like how the `propTypes` of the same component has.

### Reproduction/example

https://stackblitz.com/edit/github-sfqmlu1x?file=src%2FApp.jsx&preset=node=

### Steps to reproduce

Open the stackblitz and see the console error that says either of them is requried.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by adamalston** [comment]
Similar to https://github.com/carbon-design-system/carbon/issues/19472.

**Comment by mariyageorge01** [comment]
In my understanding, this issue is not similar to https://github.com/carbon-design-system/carbon/issues/19472

The issue mentioned in here is that we cannot have aria-label and aria-labelledby on header when header is a descendant of main. Ref: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/header#accessibility. When header is used as a descendant of main, the implicit aria role is generic and when aria role is generic, we cannot have aria-label or aria-labelledby. Ref: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/generic_role. 

In Header carbon component, either aria-label or aria-labelledby was required. If we don't add at least one of them, we get the warning:
![Image](https://github.com/user-attachments/assets/b8b65901-0bd7-48ff-a743-e16e981e8233)

But, if we add one of them, we get the accessibility violation: `The ARIA attributes "aria-label" are not valid for the element <header> with implicit ARIA role "generic`

So, the solution here is to make both aria-label and aria-labelledby optional.

See more details in https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1750333381614129

**Comment by mariyageorge01** [comment]
Can I be assigned to this issue, please?
